### ISSUE TITLE
Add support for relative compilerPath

### ISSUE THREAD

**Issue by timedroid** [microsoft/vscode-cpptools#1982]
Feature Request:
It appears compilerPath currently only supports an absolute path to the compiler.
Please add the ability to specify a path relative to ${workspaceRoot}.

**Comment by sean-mcmanus** [comment]
Should be easy to add.

**Comment by ai-chen2050** [comment]
you could temp to add "\"${workspaceRoot}\"", maybe it will work !

**Comment by raggi** [comment]
I would like this too, but I need a bit more than just the compiler path.

Starting the string with e.g. ${workspaceRoot} to get to the clang binary works ok, but I also need to specify --sysroot and --target to get correct behavior.

I would like for these settings to be split out a bit more:

* minimally compilerPath: support repeated replacement of `${workspaceFolder}` (this would enable me to make this work for e.g. a Zircon checkout).
* consider adding "target" and "sysroot" as specific options in clang configuration.

Example setting for me:

``` json
{
            "compilerPath": "${workspaceFolder}/zircon/prebuilt/downloads/clang/bin/clang++ --sysroot=${workspaceFolder}/out/build-zircon/build-x64/sysroot --target=x86_64-fuchsia",
}
```

In my working version of the above setting, I have to hardcode the --sysroot path to my workspace folder.

In the ideal case I would be able to parameterize out the x64 and x86_64 parts too.

**Comment by sean-mcmanus** [comment]
We've added the ability to use ${workspaceRoot}, ${workspaceFolder}, or ${vcpkgRoot} variables in the compilerPath for the next release...using ".." also works with those.

@raggi The sysroot/target in compile commands is being tracked in https://github.com/Microsoft/vscode-cpptools/issues/1575 . The current workaround is to add the sysroot and target options to the compilerPath, which should work unless you have files that use different settings.

**Comment by sean-mcmanus** [comment]
We're using https://github.com/Microsoft/vscode-cpptools/issues/2397 to track enabling relative paths without explicitly using ${workspaceFolder}.

**Comment by sean-mcmanus** [comment]
Fixed with 0.18.0 (requires usage of ${workspaceFolder}).
